File: lib/HttpCapture/DevHttpCaptureCompiler/src/main/java/dev/capture/base/BaseDevHttpActivity.java
package dev.capture.base;

import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.Nullable;

import dev.capture.UtilsCompiler;

/**
 * detail: DevHttpCapture Base Activity
 * @author Ttt
 */
public class BaseDevHttpActivity
        extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // 添加 Activity
        UtilsCompiler.getInstance().addActivity(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        // 移除 Activity
        UtilsCompiler.getInstance().removeActivity(this);
    }
}

File: lib/HttpCapture/DevHttpCaptureCompiler/src/main/java/dev/capture/base/BaseDevHttpActivity.kt
package dev.capture.base

import android.app.Activity
import android.os.Bundle
import androidx.viewbinding.ViewBinding
import dev.capture.UtilsCompiler

/**
 * detail: DevHttpCapture Base Activity
 * @author Ttt
 */
abstract class BaseDevHttpActivity<VB : ViewBinding> : Activity() {

    lateinit var binding: VB

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = createBinding()
        // 添加 Activity
        UtilsCompiler.addActivity(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        // 移除 Activity
        UtilsCompiler.removeActivity(this)
    }

    override fun onBackPressed() {
        finishOperate()
    }

    // ============
    // = abstract =
    // ============

    abstract fun createBinding(): VB

    abstract fun finishOperate()
}